### ISSUE TITLE
On preprocessing - keep foreign namespace elements

### ISSUE THREAD

**Issue by phax** [phax/ph-schematron#51]
As outlined in #48;
Currently the following example:
```xml
<?xml version="1.0" encoding="utf-8"?>
<sch:schema xmlns="http://purl.oclc.org/dsdl/schematron" xmlns:sch="http://purl.oclc.org/dsdl/schematron"
  queryBinding="xslt2"
  xmlns:sqf="http://example.org/sqf">
  
  <sch:ns prefix="sqf" uri="http://example.org/sqf" />
  
  <sch:pattern name="Customer">
    <sch:rule context="/file/Customer">
      <sch:assert test="base-uri()" role="fatal">
        base-uri is '<sch:value-of select="base-uri()" />'
        <sqf:fix id="a12"/>
      </sch:assert>
    </sch:rule>
  </sch:pattern>
</sch:schema> 
```
is preprocessed to 
```xml
<?xml version="1.0" encoding="UTF-8"?>
<schema xmlns="http://purl.oclc.org/dsdl/schematron" queryBinding="xslt2">
  <ns prefix="sqf" uri="http://example.org/sqf" />
  <pattern name="Customer">
    <rule context="/file/Customer">
      <assert test="base-uri()" role="fatal">
        <ns0:fix xmlns:ns0="http://example.org/sqf" id="a12" />
        base-uri is '<value-of select="base-uri()" />'
        
            </assert>
    </rule>
  </pattern>
</schema>
```
so the `<sqf:fix id="a12"/>` was moved and the namespace prefix went away...

**Comment by fviolette** [comment]
I actually declare xmlns:sqf in <schema> so this should be taken into account, if that behaves differently. I need to double check if my use or <sch:ns> is affected as well.

**Comment by phax** [comment]
The problem is the underyling data model, because the order of elements is not maintained in the `PSAssertReport` class - the foreign elements are always at the end

**Comment by phax** [comment]
Fixed in 5.0.10 - the order of foreign elements is now more flexible and not so strict. According to the RNG, foreign elements are usually only supported at the beginning of an element.
